agent/api/task/task.go

```diff
@@ -709,6 +734,92 @@ func (task *Task) getAllASMSecretRequirements() map[string]apicontainer.Secret {
 	return reqs
 }
 
+// hasLogRouter returns true if the task has a log router container.
+func (task *Task) hasLogRouter() bool {
+	for _, container := range task.Containers {
+		if container.LogRouter != nil { // This is a log router container.
+			return true
```

Comment:
it might be better to check the log router type here. But this can be refactored later as we add more log routers

Reply:
not sure if i understand correctly, but i think regardless of the log router type, hasLogRouter should return true if container.LogRouter != nil. If there are more log router types later, initializeLogRouterResource method or the log router resource itself probably needs to change to deal with new type.